virttest/libvirt_vm.py

```diff
@@ -2795,73 +2796,28 @@ def install_package(self, name, ignore_status=False):
         :param name: Name of package to be installed
         """
         session = self.wait_for_login()
-        vm_distro = self.get_distro()
-        try:
-            # distro specific support for package manager
-            if vm_distro.lower() == 'ubuntu':
-                query_cmd = "dpkg -l | grep %s" % name
-                cmd = "apt install -y %s" % name
-                update_cmd = "apt upgrade -y %s" % name
-                clean_cmd = "apt-get clean"
-            else:
-                query_cmd = "rpm -q %s" % name
-                cmd = "yum install -y %s" % name
-                update_cmd = "yum update -y %s" % name
-                clean_cmd = "yum clean all"
-
-            if session.cmd_status(query_cmd):
-                # Install the package if it does not exists
-                status, output = session.cmd_status_output(cmd, timeout=300)
-                # Just check status is not enough
-                # It's necessary to check if install successfully
-                if status != 0 or session.cmd_status(query_cmd) != 0:
-                    # The local repo database probably is broken.
-                    session.cmd_status_output(clean_cmd, timeout=300)
-                    # Retry install required packages
-                    status, output = session.cmd_status_output(cmd, timeout=300)
-                    if status != 0 or session.cmd_status(query_cmd) != 0:
-                        if not ignore_status:
-                            raise virt_vm.VMError("Installation of package %s "
-                                                  "failed:\n%s" % (name, output))
-                        else:
-                            logging.error("Installation of package %s failed:"
-                                          "\n%s", name, output)
-            else:
-                # Update the package
-                status, output = session.cmd_status_output(update_cmd,
-                                                           timeout=600)
-                if status:
-                    if not ignore_status:
-                        raise virt_vm.VMError("Update of package %s failed:"
-                                              "\n%s" % (name, output))
-                    else:
-                        logging.error("Update of package %s failed:"
-                                      "\n%s", name, output)
-        finally:
-            session.close()
+        if not utils_package.package_install(name, session):
+            if not ignore_status:
+                session.close()
```

Comment:
session.close() call twice in line 2801 and 2805,actually can merge call once if put it before line 2800

Reply:
we cant keep session.close() in 2800, then session will close only on utils_package fails. I think it is in right place.